### PR TITLE
Fix #6 No support for top-level scenes and scene properties 

### LINKED ISSUE #6: No support for top-level `scenes` and `scene` properties
GlTF defines an optional field `scenes`, which is a list of scenes where each scene contains a name and a list of the root nodes (from the top-level `nodes` property) that belong to it.

The specification also defines an optional `scene` field which is the index of the default scene for this GlTF file.

A scene object is at least useful in practice to find the root nodes of the scene hierarchy before traversing a model defined in a GlTF file.

NB: Currently the package dependency `gltf-codec` only supports `scenes` but I've submitted a patch for `scene` field support.

### PR DESCRIPTION
**Pull request by mrehayden1**

### CODE REVIEW

**Comment by sgillespie** [comment]
This breaks some tests: https://github.com/sgillespie/haskell-gltf-loader/actions/runs/7993144849/job/21874943312

File: src/Text/GLTF/Loader/Gltf.hs [hunk 1]
```diff
@@ -14,6 +14,7 @@ module Text.GLTF.Loader.Gltf
     MeshPrimitiveMode(..),
     Sampler(..),
     SamplerWrap(..),
+    Scene(..),
```
Comment (sgillespie) [inline src/Text/GLTF/Loader/Gltf.hs:17]: I would also like to have a lens for this in here

Comment (mrehayden1) [reply src/Text/GLTF/Loader/Gltf.hs:17]: Will do

**Comment by mrehayden1** [comment]
Okay, rebased onto main and with added spec tests and lenses.

**Pull request merged**